### ISSUE TITLE
mvn -Pprod creates invalid CSS with semi-colons

### ISSUE THREAD

**Issue by bluegaspode** [jhipster/generator-jhipster#3906]
##### **Overview of the issue**
mvn -Pprod creates invalid CSS files with semi-colons between the concatenated files.

As far as I can see, #3020 (originating from #3017) introduces the issue.

##### **Motivation for or Use Case**
In my environment I was using font-awesome, which worked fine in dev mode but not at all in production. The fonts were not loaded. 
After debugging and comparing CSS files a lot, finally I realized, that in target/www/content/css/vendor-ZZZ.css a semicolon between all css files was introduced.
This seems to hide the very first css declaration of any css file concatenated in production mode.

##### **Reproduce the error**
- generate a new jhipster project with default values
- to simplify debugging prevent cssnano from running by removing the line 
`   .pipe(gulpIf('*.css', cssTask()))`
in /gulp/build.js
- run mvn -Pprod
- inspect target/www/content/css/vendor-ZZZ.css
- between the bootstrap.css and angular-loading-bar.css you will find an invalid semicolon
This is how it looks in my environment (last line from bootstrap.css, semicolon, beginning of angular-loading-bar.css):

```
/*# sourceMappingURL=bootstrap.css.map */
;
/*! 
 * angular-loading-bar v0.9.0
 * https://chieffancypants.github.io/angular-loading-bar
 * Copyright (c) 2016 Wes Cruver
 * License: MIT
 */
```

##### **Related issues**
#3020 (originating from #3017)

##### **Suggest a Fix**
removing the line 
`    .pipe(footer, ';');`
in ./gulp/build.js fixes my problem, though I don't don't have the same setup as in the original issue (I'm not using I18n). 
I'm not an expert in gulp, maybe there is a way to add the semicolon only to *.js files, but not to *.css files?

##### **JHipster Version(s)**
found in 3.4.2 and reproduced as well in 3.5.1

**Comment by deepu105** [comment]
Thanks a lot for getting to the bottom of this. I guess we need to do the
footer pipe only for scripts. Ill take a look

Thanks & regards,
Deepu

**Comment by deepu105** [comment]
Ok I did some tests. So actually current usage of footer to add `;` seems to be working  only on css files and the js files are not affected, the js files seems to have `,` added and I guess `useref` is taking care of that. So adding `;` doesnt make any difference so im removing it [here](https://github.com/jhipster/generator-jhipster/commit/61a281befed44346880a407bc78f185fd3cf335f). The original issue #3017 was when we were using `usemin` but then we switched to `useref` down the line.
@erikkemperman you reported the original issue hence would like to hear your thoughts as well

@bluegaspode please check if this works for your usecase

**Comment by erikkemperman** [comment]
The footer was added specifically to address concatenation of js by usemin. Not sure why it ended up being used for css? Anyway, since usemin is out of the picture it's probably allright to get rid of the footer. Note though the original problem only became apparent in config with multiple languages, so maybe double check that case still works.

**Comment by bluegaspode** [comment]
Hi @deepu105 , 
currently I don't now how to test with the master of JHipster. Anyway: your change is exactly what I changed in my local environment and I'm happy to see it applied officially.

Cheers
Stefan

**Comment by gmarziou** [comment]
It's documented here : https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md#setup

**Comment by yogeshkrishnani** [comment]
I am facing the same issue. 

mvn -Pprod creates CSS files with semi-colons between the files.

**Comment by gmarziou** [comment]
@yogeshkrishnani Which version? Do you mean that the fix did not work for you?

**Comment by yogeshkrishnani** [comment]
@gmarziou The fix did work for me. I am using JHipster version 3.1.0.

**Comment by deepu105** [comment]
then why are you commenting here?


Thanks & Regards,
Deepu

On Fri, Oct 28, 2016 at 11:28 AM, Yogesh Krishnani <notifications@github.com

**Comment by yogeshkrishnani** [comment]
Just to tell, Other users are also affected from it. Nothing else.

**Comment by gmarziou** [comment]
This fix has been released in version 3.6.0 (current one is 3.9.1), so no other users will be affected except the ones like you using an old version. 

When an issue is closed, we don't expect people to comment it further unless the fix is not working.

**Comment by yogeshkrishnani** [comment]
Okay I understand now.
